src/tests/functional/inference_engine/CMakeLists.txt

```diff
@@ -69,18 +69,22 @@ addIeTargetTest(
         IE
 )
 
+target_compile_definitions(${TARGET_NAME} PRIVATE $<$<BOOL:${BUILD_SHARED_LIBS}>:"BUILD_SHARED_LIBS">)
```

Comment:
it looks like this line causes this issue on Ubuntu during cmake loading with this flag -DENABLE_TESTS:BOOL=ON

Compiler exited with error code 1: /usr/bin/c++
	<command-line>: error: macro names must be identifiers
	/tmp/compiler-file13789279785052064369:27:5: warning: "__clang__" is not defined, evaluates to 0 [-Wundef]
	   27 | #if __clang__
	      |     ^~~~~~~~~


cmd line:
cmake -DCMAKE_BUILD_TYPE=Debug -DENABLE_TESTS:BOOL=ON -DENABLE_HETERO:BOOL=OFF -DENABLE_PYTHON:BOOL=ON -DENABLE_MULTI:BOOL=OFF -DENABLE_ONEDNN_FOR_GPU:BOOL=OFF -DENABLE_CLDNN:BOOL=OFF -DENABLE_AUTO_BATCH:BOOL=OFF -DENABLE_AUTO:BOOL=OFF -DENABLE_INTEL_GPU:BOOL=OFF -DENABLE_INTEL_MYRIAD:BOOL=OFF -DENABLE_INTEL_MYRIAD_COMMON:BOOL=OFF -DENABLE_IR_V7_READER:BOOL=OFF -DENABLE_OV_PADDLE_FRONTEND:BOOL=OFF -S /home/itikhonov/OpenVINO/openvino -B /home/itikhonov/OpenVINO/openvino/cmake-build-debug

system info:
-- The C compiler identification is GNU 11.2.0
-- The CXX compiler identification is GNU 11.2.0
OS: Ubuntu 20.04 or 22.04

Reply:
I've check this cmake command on Ubuntu 20 compilers gcc 9.4, 11.1 (don't have 11.2) and same cmake version ( and one older version) no issue on cmake configuration step.
Is there any issue on CI check?
I can't reproduce it.

Reply:
for some reason it's not reproducible in CI, only locally but on 2 machines 
please check that you have BUILD_SHARED_LIBS=ON, but actually it should be ON by default